plugins/ccgx/fu-ccgx-dmc-device.c

```diff
@@ -616,33 +620,32 @@ fu_ccgx_dmc_device_attach(FuDevice *device, FuProgress *progress, GError **error
 		return TRUE;
 
 	if (self->update_model == DMC_UPDATE_MODEL_DOWNLOAD_TRIGGER) {
-		DmcTriggerCode trigger_code = DMC_TRIGGER_CODE_UPDATE_NOW;
-
-		if (manual_replug)
-			trigger_code = DMC_TRIGGER_CODE_UPDATE_ON_DISCONNECT;
-
-		if (!fu_ccgx_dmc_device_send_download_trigger(self, trigger_code, error)) {
-			g_prefix_error(error, "download trigger error: ");
-			return FALSE;
+		if (self->trig_code > 0) {
+			if (!fu_ccgx_dmc_device_send_download_trigger(self,
+								      self->trig_code,
+								      error)) {
+				g_prefix_error(error, "download trigger error: ");
+				return FALSE;
+			}
 		}
 	} else if (self->update_model == DMC_UPDATE_MODEL_PENDING_RESET) {
 		if (!fu_ccgx_dmc_device_send_sort_reset(self, manual_replug, error)) {
 			g_prefix_error(error, "soft reset error: ");
 			return FALSE;
 		}
-	} else {
-		g_set_error(error,
-			    FWUPD_ERROR,
-			    FWUPD_ERROR_NOT_SUPPORTED,
-			    "invalid update model = %u",
-			    self->update_model);
-		return FALSE;
 	}
 
-	if (manual_replug)
+	if (manual_replug) {
+		fu_device_add_flag(device, FWUPD_DEVICE_FLAG_NEEDS_ACTIVATION);
```

Comment:
Is there a way to actually trigger update without disconnect too after staged?

If so this flag makes sense but you need to add another vfunc to handle sending that separately if requested by the client. I can show you an example from another plugin if so.

If it's not possible (or desirable) then please remove this flag and only use the inhibition.